### PR TITLE
[Docs] Ingestion engines - sources/targets compatibility matrix

### PR DESCRIPTION
**Pull request by assaf758**
[ML-3424](https://jira.iguazeng.com/browse/ML-3424)

### CODE REVIEW

File: docs/serving/available-steps.md [hunk 1]
```diff
@@ -73,26 +73,30 @@ FeaeturesetValidator | Y      | N     | Y      |
...
-| [mlrun.datastore.StreamSource](../api/mlrun.datastore.html#mlrun.datastore.StreamSource) | Sets the stream source for the flow. If the stream doesn’t exist it creates it.  | 
-
+| Class name                                                                                                        | Description                                                                     | storey | spark | pandas |
+| --------------------------------------------------                                                                | ---------------------------------                                               | ---    | ---   | ---    |
+| [mlrun.datastore.BigQuerySource](../api/mlrun.datastore.html#mlrun.datastore.BigQuerySource)                      | Reads Google BigQuery query results as input source for a flow.                 | N      | Y     | Y      |
+| [mlrun.datastore.SnowFlakeSource]                                                                                 | Reads SnowFlak query results as input source for a flow.                        | N      | Y     | N      |
```
Comment (gtopper) [inline docs/serving/available-steps.md:79]: Missing a letter, plus apparently the F in Snowflake is not capitalized.
```suggestion
| [mlrun.datastore.SnowFlakeSource]                                                                                 | Reads Snowflake query results as input source for a flow.                        | N      | Y     | N      |
```

File: docs/serving/available-steps.md [hunk 2]
```diff
@@ -73,26 +73,30 @@ FeaeturesetValidator | Y      | N     | Y      |
...
+| Class name                                                                                                                | Description                                                                                                  | storey | spark | pandas |
+| --------------------------------------------------                                                                        | -------------------------------------------------------                                                      | ---    | ---   | ---    |
+| [mlrun.datastore.CSVTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.CSVTarget)                    | Writes events to a CSV file.                                                                                 | Y      | Y     | Y      |
+| [mlrun.datastore.NoSqlTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.NoSqlTarget)                | Persists the data in V3IO *table* to its associated storage by key.                                          | Y      | Y     | Y      |
```
Comment (gtopper) [inline docs/serving/available-steps.md:92]: I know it was like that before, but any idea why "table" is in italics?

File: docs/serving/available-steps.md [hunk 3]
```diff
@@ -73,26 +73,30 @@ FeaeturesetValidator | Y      | N     | Y      |
...
+| [mlrun.datastore.SqlTarget]                                                                                               | Persists the data in SQL *table* to its associated storage by key.                                           | Y      | N     | Y      |
+| [mlrun.datastore.ParquetTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.ParquetTarget)            | The Parquet target storage driver, used to materialize feature set/vector data into parquet files.           | Y      | Y     | Y      |
+| [mlrun.datastore.StreamTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.StreamTarget)              | Writes all incoming events into a V3IO stream.                                                               | Y      | N     | N      |
+| [mlrun.datastore.KafkaTarget]                                                                                             | Writes all incoming events into a Kafka stream.                                                              | Y      | N     | N      |
```
Comment (gtopper) [inline docs/serving/available-steps.md:97]: Looks like the import needs to be fixed before we document this.
```
>>> from mlrun.datastore import KafkaTarget
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'KafkaTarget' from 'mlrun.datastore' (/Users/gal/work/mlrun/mlrun/datastore/__init__.py)
```

File: docs/serving/available-steps.md [hunk 4]
```diff
@@ -73,26 +73,30 @@ FeaeturesetValidator | Y      | N     | Y      |
...
+| [storey.transformations.ToDataFrame](https://storey.readthedocs.io/en/latest/api.html#storey.transformations.ToDataFrame) | Create pandas data frame from events. Can appear in the middle of the flow, as opposed to ReduceToDataFrame. | Y      | N     | N      |
+| [mlrun.datastore.TSBDTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.TSDBTarget)                  | Writes incoming events to TSDB table.                                                                        | Y      | N     | Y      |
```
Comment (gtopper) [inline docs/serving/available-steps.md:99]: I know it was there before but...
```suggestion
| [mlrun.datastore.TSDBTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.TSDBTarget)                  | Writes incoming events to TSDB table.                                                                        | Y      | N     | Y      |
```
And even then, the import is broken for it as well:
```
>>> from mlrun.datastore import TSDBTarget
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'TSDBTarget' from 'mlrun.datastore' (/Users/gal/work/mlrun/mlrun/datastore/__init__.py)
```

File: docs/serving/available-steps.md [hunk 5]
```diff
@@ -73,26 +73,30 @@ FeaeturesetValidator | Y      | N     | Y      |
...
+| [mlrun.datastore.ParquetTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.ParquetTarget)            | The Parquet target storage driver, used to materialize feature set/vector data into parquet files.           | Y      | Y     | Y      |
+| [mlrun.datastore.StreamTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.StreamTarget)              | Writes all incoming events into a V3IO stream.                                                               | Y      | N     | N      |
+| [mlrun.datastore.KafkaTarget]                                                                                             | Writes all incoming events into a Kafka stream.                                                              | Y      | N     | N      |
+| [storey.transformations.ToDataFrame](https://storey.readthedocs.io/en/latest/api.html#storey.transformations.ToDataFrame) | Create pandas data frame from events. Can appear in the middle of the flow, as opposed to ReduceToDataFrame. | Y      | N     | N      |
```
Comment (gtopper) [inline docs/serving/available-steps.md:98]: I know it was there before, but it's kind of strange that this references a step that is not documented on the list (`ReduceToDataFrame`).

Comment (assaf758) [reply docs/serving/available-steps.md]: removed the reference

File: docs/serving/available-steps.md (hunk 4 above)
Comment (assaf758) [reply docs/serving/available-steps.md]: removed (anyway TSDB is on the way out)

File: docs/serving/available-steps.md (hunk 3 above)
Comment (assaf758) [reply docs/serving/available-steps.md]: removed for now.

File: docs/serving/available-steps.md (hunk 2 above)
Comment (assaf758) [reply docs/serving/available-steps.md]: removed the italic

File: docs/serving/available-steps.md (hunk 1 above)
Comment (assaf758) [reply docs/serving/available-steps.md]: fixed

**Review by gtopper — APPROVED** [review]
👍🏻

**Pull request merged**